google-cloud-storage/src/main/scala/akka/stream/alpakka/googlecloud/storage/impl/GCStorageStream.scala

```diff
@@ -142,36 +136,41 @@ import scala.util.control.NonFatal
       .mapMaterializedValue(_ => NotUsed)
   }
 
-  def getObject(bucket: String, objectName: String): Source[Option[StorageObject], NotUsed] =
+  def getObject(bucket: String, objectName: String, generation: Option[Long]): Source[Option[StorageObject], NotUsed] =
     Setup
       .source { implicit mat => implicit attr =>
         makeRequestSource(
           createRequestSource(
             uriFactory = (settings: GCStorageSettings) =>
-              Uri(settings.baseUrl).withPath(Path(settings.basePath) ++ getObjectPath(bucket, objectName))
+              Uri(settings.baseUrl)
+                .withPath(Path(settings.basePath) ++ getObjectPath(bucket, objectName))
+                .withQuery(Query(generation.map("generation" -> _.toString).toMap))
           )
         ).mapAsync(parallelism)(
           response => processGetStorageObjectResponse(response, mat)
         )
       }
       .mapMaterializedValue(_ => NotUsed)
 
-  def deleteObjectSource(bucket: String, objectName: String): Source[Boolean, NotUsed] =
+  def deleteObjectSource(bucket: String, objectName: String, generation: Option[Long]): Source[Boolean, NotUsed] =
     Setup
       .source { implicit mat => implicit attr =>
         makeRequestSource(
-          createRequestSource(HttpMethods.DELETE,
-                              uriFactory = (settings: GCStorageSettings) =>
-                                Uri(settings.baseUrl)
-                                  .withPath(Path(settings.basePath) ++ getObjectPath(bucket, objectName)))
+          createRequestSource(
+            HttpMethods.DELETE,
+            uriFactory = (settings: GCStorageSettings) =>
+              Uri(settings.baseUrl)
+                .withPath(Path(settings.basePath) ++ getObjectPath(bucket, objectName))
+                .withQuery(Query(generation.map("generation" -> _.toString).toMap))
+          )
         ).mapAsync(parallelism)(entityForSuccessOption)
           .map(option => option.isDefined)
       }
       .mapMaterializedValue(_ => NotUsed)
 
   def deleteObjectsByPrefixSource(bucket: String, prefix: Option[String]): Source[Boolean, NotUsed] =
-    listBucket(bucket, prefix)
-      .flatMapConcat(listBucketResult => deleteObjectSource(bucket, listBucketResult.name))
+    listBucket(bucket, prefix, versions = false)
+      .flatMapConcat(listBucketResult => deleteObjectSource(bucket, listBucketResult.name, None))
```

Comment:
Might be good to use `generation = None` for these.

Reply:
I also thought of that, but wouldn't it break the binary compatibility?
or we can just add this to exclusion rules, since those are parts of `InternalApi`?

Reply:
I mean to add the parameter name at use-side, that won't change the API. But yes, `impl` is automatically excluded from Mima checks as it is always considered internal API.